File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isValidBST(TreeNode root) {
        List<Integer> ans = new ArrayList<>();
        isValidBSTHelper(root,ans);
        for(int i = 1;i<ans.size();i++){
            if(ans.get(i)<=ans.get(i-1)){
                return false;
            }
        }
        return true;
    }
    public void isValidBSTHelper(TreeNode root, List<Integer> inorder){
        if(root == null) return;
        isValidBSTHelper(root.left, inorder);
        inorder.add(root.val);
        isValidBSTHelper(root.right, inorder);
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        ans = []
        queue = deque()
        if root is not None:
            queue.append(root)
        while queue:
            length = len(queue)
            level = []
            for _ in range(length):
                current = queue.popleft()
                level.append(current.val)
                if current.left is not None:
                    queue.append(current.left)
                if current.right is not None:
                    queue.append(current.right)
            ans.append(level)
        return ans

        